task-list/app/views/tasks/new.html.erb

```diff
@@ -1,6 +1,7 @@
-<%= form_for @task do |f| %>
+<%= form_for @task, url: { action: @action}, method: @method do |f| %>
```

Comment:
Great use of variables from the controller to make this template work for both `new` and `edit`!